template/src/ontology/Makefile.jinja2

```diff
@@ -829,28 +830,52 @@ $(PATTERNDIR)/dospd-patterns/%.yml: download_patterns
 # DOSDP default pipeline
 
 DOSDP_TSV_FILES_DEFAULT = $(wildcard $(PATTERNDIR)/data/default/*.tsv)
-DOSDP_OWL_FILES_DEFAULT = $(patsubst %.tsv, $(PATTERNDIR)/data/default/%.ofn, $(notdir $(wildcard $(PATTERNDIR)/data/default/*.tsv)))
-DOSDP_TERM_FILES_DEFAULT = $(patsubst %.tsv, $(PATTERNDIR)/data/default/%.txt, $(notdir $(wildcard $(PATTERNDIR)/data/default/*.tsv)))
-DOSDP_PATTERN_NAMES_DEFAULT = $(strip $(patsubst %.tsv,%, $(notdir $(wildcard $(PATTERNDIR)/data/default/*.tsv))))
+DOSDP_PATTERN_NAMES_DEFAULT = $(strip $(patsubst %.tsv, %, $(notdir $(DOSDP_TSV_FILES_DEFAULT))))
+DOSDP_OWL_FILES_DEFAULT = $(foreach name, $(DOSDP_PATTERN_NAMES_DEFAULT), $(PATTERNDIR)/data/default/$(name).ofn)
+DOSDP_TERM_FILES_DEFAULT = $(foreach name, $(DOSDP_PATTERN_NAMES_DEFAULT), $(PATTERNDIR)/data/default/$(name).txt)
+DOSDP_YAML_FILES_DEFAULT = $(foreach name, $(DOSDP_PATTERN_NAMES_DEFAULT), $(PATTERNDIR)/dosdp-patterns/$(name).yaml)
 
 $(DOSDP_OWL_FILES_DEFAULT): $(EDIT_PREPROCESSED) $(DOSDP_TSV_FILES_DEFAULT) $(ALL_PATTERN_FILES)
 	if [ $(PAT) = true ] && [ "${DOSDP_PATTERN_NAMES_DEFAULT}" ]; then $(DOSDPT) generate --catalog=$(CATALOG) \
     --infile=$(PATTERNDIR)/data/default/ --template=$(PATTERNDIR)/dosdp-patterns --batch-patterns="$(DOSDP_PATTERN_NAMES_DEFAULT)" \
     --ontology=$< {{ project.dosdp_tools_options }} --outfile=$(PATTERNDIR)/data/default; fi
 
+.PHONY: dosdp-docs-default
+dosdp-docs-default: $(EDIT_PREPROCESSED) $(DOSDP_TSV_FILES_DEFAULT) $(DOSDP_YAML_FILES_DEFAULT)
+	mkdir -p $(DOCSDIR)/patterns/default
+	$(DOSDPT) docs {{ project.dosdp_tools_options }} --catalog=$(CATALOG) \
+		       --ontology=$< \
+		       --infile=$(PATTERNDIR)/data/default \
+		       --template=$(PATTERNDIR)/dosdp-patterns \
+		       --batch-patterns="$(DOSDP_PATTERN_NAMES_DEFAULT)" \
+		       --outfile=$(DOCSDIR)/patterns/default{% if project.github_org and project.repo %} \
+		       --data-location-prefix=https://github.com/{{ project.github_org }}/{{ project.repo }}/tree/{{ project.git_main_branch }}/src/patterns/data/default{% endif %}
```

Comment:
Just noting this is web location not "raw" which is probably correct for docs, and it is a bit inconvenient for our gitlab users like @ehartley to have GitHub assumptions hard coded into this - maybe just factor this into a make variable to make it  at least overwritable at runtime?

Reply:
> Just noting this is web location not "raw" which is probably correct for docs

Yes. For context, what this `--data-location-prefix` option does is providing the link that `dosdp-tools docs` will insert in the documentation, along with a message saying something like “to see the full TSV table for this pattern, please go there”.

> maybe just factor this into a make variable to make it at least overwritable at runtime?

Note that this is already conditional to the ODK configuration defining the `github_org` variable. For a project hosted elsewhere than on GitHub, the `--data-location-prefix` option will be skipped entirely.

Since we already have `github_org` and `repo` in the ODK configuration file, I think that if we want to enable this feature for non-GitHub-hosted repos, it should also happen in the ODK configuration rather than in the Makefile.

And since we cannot obviously handle all possible repository hosting services, I would suggest adding a single `repo_url` (or similar) configuration option, to be used whenever `github_org` is not set.

So, a GitHub-hosted project would define:

```
repo: my-repo-name
github_org: my-organisation
```

which would yield the `--data-location-prefix` URL shown above, while a non-GitHub-hosted project would define:

```
repo: my-repo-name
repo_url: https://my.custom.hosting.provider.example.net/root/of/my/repository
```

which would yield `--data-location-prefix=https://my.custom.hosting.provider.example.net/root/of/my/repository/src/patterns/data/default/`.

Reply:
I checked the rest of the ODK setup and `github_org` is connected to soo many assumptions on GitHub that I have to agree with you (I was going to say that gitlab has indeed quite a similar structure and this specific feature _could_ work for gitlab at least, but as you say, it would not support any other hosting infrastructure).

Cool lets proceed then. The github_org refactor is massive so I don't think its worth trying to do this here as part of this specific PR unless you want to just start for this feature alone!

Reply:
> I don't think its worth trying to do this here as part of this specific PR unless you want to just start for this feature alone!

Well, we have to start somewhere :) , so yes, if you agree with the idea of a generic `repo_url` config key I’ll amend this PR to add it.

Reply:
Totally agreed!